saos-import/src/main/java/pl/edu/icm/saos/importer/notapi/common/content/transaction/FilesOperationsTransaction.java

```diff
@@ -0,0 +1,75 @@
+package pl.edu.icm.saos.importer.notapi.common.content.transaction;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.LinkedList;
+import java.util.Queue;
+
+import org.apache.commons.io.FileUtils;
+
+import com.google.common.io.Files;
+
+/**
+ * Handler providing functionality to perform transaction like operations on files.
+ * It can perform series of adding and deleting files within destination directory.
+ * Next these operations can be commited or rollbacked
+ * (using {@link FilesOperationsTransactionManager#commit(String)} or {@link FilesOperationsTransactionManager#rollback(String)}).
+ * Commits approves all changes and applies it to destination directory.
+ * Rollbacks withdraws all changes and leaves destination directory in a state
+ * as it was at the beginning.
+ * 
+ * This class uses internally some temporary directory to store files that will be deleted by commit operation.
+ * 
+ * @author madryk
+ */
+public class FilesOperationsTransaction {
+    
+    private File deletedTmpDirectory;
+    
+    private Queue<String> addedFilesList;
+    
+    private File contentDirectory;
+    
+    
+    //------------------------ CONTRUCTORS --------------------------
+    
+    FilesOperationsTransaction(File contentDirectory) {
+        this.contentDirectory = contentDirectory;
+        
+        deletedTmpDirectory = Files.createTempDir();
+        addedFilesList = new LinkedList<String>();
+    }
+    
+    
+    //------------------------ LOGIC --------------------------
+    
+    public void addFile(InputStream inputStream, String path) throws IOException {
```

Comment:
I don't think that it's a good idea for the 'transaction' to have service methods like addFile or overwriteFile.